### DISCUSSION TITLE
Realtime API - Everything returns empty result object

### DISCUSSION THREAD

**Discussion by computerquip** [RocketChat/Rocket.Chat#33680] [Q&A]
Here's an example exchange with `wss://open.rocket.chat/websocket`:
```
<== '{"server_id":"0"}'
==> '{"msg":"connect","version":"1","support":["1"]}'
<== '{"msg":"ping"}'
==> '{"msg":"pong"}'
<== '{"msg":"connected","session":"330328f0-8eb3-11ef-9bce-f1103112ae1f"}'
==> '{"msg":"method","method":"login","id":"0","params":[{"resume":"<snip>"}]}'
<== '{"msg":"result","id":"0","result":{"id":"ga33SsqjhgYSxHQrc","token":"<snip>","type":"resume"}}'
==> '{"msg":"method","method":"getUserRoles","id":"1","params":[]}'
<== '{"msg":"updated","methods":["0"]}'
failed to process rocketchat message: unknown message type 'updated', skipping
<== '{"msg":"result","id":"1","result":{}}'
<== '{"msg":"updated","methods":["1"]}'
failed to process rocketchat message: unknown message type 'updated', skipping
==> '{"msg":"method","method":"checkUsernameAvailability","id":"2","params":["henry103433"]}'
<== '{"msg":"result","id":"2","result":{}}'
<== '{"msg":"updated","methods":["2"]}'
failed to process rocketchat message: unknown message type 'updated', skipping
==> '{"msg":"method","method":"asdf","id":"3","params":[]}'
<== '{"msg":"result","id":"3","result":{}}'
<== '{"msg":"updated","methods":["3"]}'
failed to process rocketchat message: unknown message type 'updated', skipping
```
My primary question is: Why are the method calls returning empty result objects?

What is `'{"server_id":"0"}'`? This doesn't seem to be documented anywhere. This broke a few libraries I tried initially.
What are the `updated` messages? They also don't seem to be documented.

Any help would be appreciated.

**Comment by reetp** [comment]
Your best place for support is here:

**https://open.rocket.chat/channel/support**

I'd have a read here first on how to ask questions that will get answers.

https://gist.github.com/reetp/a66149d5f060f260643a353ca7067a98

https://gist.github.com/reetp/b0ba4e3d0ea2ff48ca9da00a5a647d42


I'd also go back to the docs and try the test code there.

https://developer.rocket.chat/docs/realtime-api

https://developer.rocket.chat/docs/realtime-api#resources
